lib/modules/datasource/nuget/v3.ts

```diff
@@ -212,6 +223,20 @@ export async function getReleases(
       if (sourceUrl) {
         dep.sourceUrl = massageUrl(sourceUrl);
       }
+    } else if (latestNupkgUrl) {
+      const sourceUrl = await getSourceUrlFromNupkgCached(
+        http,
+        pkgName,
+        latestStable,
+        latestNupkgUrl,
+      );
+      if (sourceUrl) {
+        dep.sourceUrl = massageUrl(sourceUrl);
+        logger.debug(
+          { sourceUrl, nupkgUrl: latestNupkgUrl },
+          `Determined sourceUrl from nupkgUrl`,
+        );
```

Comment:
```suggestion
        logger.debug(
          `Determined sourceUrl ${sourceUrl} from ${latestNupkgUrl}`,
        );
```

Reply:
Can you explain why you prefer this instead of keeping the data structured?

Reply:
I checked the code base again and the approach is a bit inconsistent. I will adapt this line as suggested, but maybe we want to discuss the desired direction and document it in https://github.com/renovatebot/renovate/tree/main/docs/development?

Reply:
Logging is inconsistent but my approach is to minimize metadate use in debug messages because it blows out the number of lines in logs. Better to use them only when the data is large, or needed for "machine" parsing (e.g. warnings, errors sent to log systems)

Reply:
@rarkins OK thanks, do you think this is worth to add to the development docs? I can start something, would need a lot of input though.

Reply:
Definitely. Main thing is that we want _no_ templating in warnings and errors (so they can be aggregated and monitored) but usually templating in debug

Reply:
OK, I have created https://github.com/renovatebot/renovate/pull/28162 to improvement the existing documentation I have just found :)